packages/features/ee/impersonation/components/ImpersonatingBanner.tsx

```diff
@@ -8,17 +9,32 @@ export type ImpersonatingBannerProps = { data: SessionContextValue["data"] };
 function ImpersonatingBanner({ data }: ImpersonatingBannerProps) {
   const { t } = useLocale();
 
-  if (!data?.user.impersonatedByUID) return null;
+  if (!data?.user.impersonatedBy) return null;
+  const returnToId = data.user.impersonatedBy.id;
+
+  const canReturnToSelf = data.user.impersonatedBy.role == "ADMIN" || data.user?.org?.id;
 
   return (
     <>
       <TopBanner
         text={t("impersonating_user_warning", { user: data.user.username })}
         variant="warning"
         actions={
-          <a className="border-b border-b-black" href="/auth/logout">
-            {t("impersonating_stop_instructions")}
-          </a>
+          canReturnToSelf ? (
```

Comment:
Validation also happens on BE just easier to handle this signout functionality here than within the custom provider from next-auth